src/main/java/hudson/plugins/ansicolor/ColorConsoleAnnotator.java

```diff
@@ -52,24 +54,33 @@
 
     private static final long serialVersionUID = 1;
 
-    private final String colorMapName;
+    private final @CheckForNull String colorMapName;
     private final @Nonnull List<AnsiAttributeElement> openTags;
 
-    ColorConsoleAnnotator(String colorMapName, @Nonnull List<AnsiAttributeElement> openTags) {
+    private ColorConsoleAnnotator(String colorMapName, @Nonnull List<AnsiAttributeElement> openTags) {
         this.colorMapName = colorMapName;
         this.openTags = openTags;
         LOGGER.log(Level.FINE, "creating annotator with colorMapName={0} openTags={1}", new Object[] { colorMapName, openTags });
     }
 
-    ColorConsoleAnnotator(String colorMapName) {
-        this(colorMapName, Collections.emptyList());
+    ColorConsoleAnnotator() {
+        this(null, Collections.emptyList());
     }
 
     @Override
     public ConsoleAnnotator<Object> annotate(@Nonnull Object context, MarkupText text) {
+        String actualColorMapName;
+        if (colorMapName == null) {
+            actualColorMapName = colorMapNameFor(context);
+            if (actualColorMapName == null) {
+                return this;
+            }
```

Comment:
I'm not sure, but I think the reason that a lot of tests are failing is that `colorMapNameFor(context)` returns null if the build has no `ColorizedAction` (same as previous behavior) or if it does have a `ColorizedAction` but the `colorMapName` is null (different from previous behavior). Previously, in the latter case, the default color map would have been used instead. Maybe it could be fixed by making `colorMapNameFor` return `AnsiColorMap` instead of just the color map name since it can distinguish between the two cases internally.